addons/purchase_stock/models/res_partner.py

```diff
@@ -22,7 +22,11 @@ def _compute_on_time_rate(self):
             ('date_order', '>', fields.Date.today() - timedelta(365)),
             ('qty_received', '!=', 0),
             ('order_id.state', 'in', ['done', 'purchase'])
-        ]).filtered(lambda l: l.product_id.sudo().product_tmpl_id.type != 'service')
+        ])
+        order_lines = self.env['purchase.order.line'].sudo().search([
```

Comment:
Nice. I initially removed this filtered but we were afraid of breaking https://github.com/odoo/odoo/commit/a08ba4be6518dd99ce93a803724165f62cbef032. I didn't think of using sudo before the search :sweat_smile: .